Review some distance analysis algorithms (#3453)
qgis/QGIS-Documentation: source/docs/user_manual/processing_algs/qgis/vectoranalysis.rst

Distance matrix
---------------
Calculates for point features distances to their nearest features in the same layer
or in another layer.

``Default menu``: :menuselection:`Vector --> Analysis Tools`

Parameters
..........

``Input point layer`` [vector: point]

>   Vector layer for which the distance matrix is calculated.

Reviewer: ```suggestion
  Point layer for which the distance matrix is calculated (**from** points).
```

Author: It took me a couple of reading to understand what you are suggesting. So I wonder if it'll be obvious to people that don't know what the alg should do. I don't have a formulation right now but how about using source/origin and destination references in the sentence instead?

Reviewer: Whatever you think is best.  My point is that it will be useful to have some more information here. The logic is that you calculate the distance **from** one point **to** another point.

``Input unique ID field`` [tablefield: any]
  Field to use to uniquely identify features of the input layer.
  Used in the output attribute table.

``Target point layer`` [vector: point]

>   Vector layer containing the nearest point(s) to search.

Reviewer: ```suggestion
  Point layer containing the nearest point(s) to search (**to** points).
```

``Target unique ID field`` [tablefield: any]
  Field to use to uniquely identify features of the target layer.
  Used in the output attribute table.

``Output matrix type`` [enumeration]
  Different types of calculation are available:

>   * 0 --- Linear (N*k x 3) distance matrix: for each input point, reports

Reviewer: ```suggestion
  * 0 --- Linear (N * _k_ x 3) distance matrix: for each input point, reports
```

>     the distance to each of its target points. Each point is output as

Reviewer: ```suggestion
    the distance to each of the _k_ nearest target points. The output matrix consists of up to _k_ rows per input point (one for each of the _k_ nearest neighbours), and each row has three columns: InputID, TargetID and Distance.
```

Author: I think we can remove `(one for each of the _k_ nearest neighbours)` as it should be obvious and, unless k is upper than target layer features, you get exactly k times the input features.
```suggestion
    the distance to each of the *k* nearest target points. The output matrix consists
    of *k* rows per input point, and each row has three columns: *InputID*, *TargetID*
    and *Distance*.
```

Reviewer: "up to" is more correct.  You will not get *k* rows per input point if there are less than *k* features to relate to.

>     many as target candidates.

Reviewer: ```suggestion
```

>   * 1 --- Standard (N x T) distance matrix

Author: Not sure of the behavior of this one. See https://issues.qgis.org/issues/21347.

  * 2 --- Summary distance matrix (mean, std. dev., min, max): for each input
    point, reports statistics on the distances to its target points.

  Default: *0*

``Use only the nearest (k) target points`` [number]
  You can choose to calculate the distance to all the points in the target layer

>   or limit to a number of closest features.

Reviewer: ```suggestion
  or limit to a number (_k_) of closest features.
```

  Default: *0* --- all the points are used.

Outputs
.......

``Distance matrix`` [vector: point]

>   Point vector layer containing the distance calculation for each input feature.

Reviewer: ```suggestion
  Point (or MultiPoint in the "Linear (N * _k_ x 3)" case)  vector layer containing the distance calculation for each input feature.
```

Reviewer: In QGIS 3.4.4, the geometry of the returned layer is a multi-geometry (reported as "MultiPoint") for the "Linear (N * _k_ x 3)" case - don't ask me why. Try it yourself.

Author: I did'nt try but I trust.

>   Its attribute table depends on the selected matrix type.

Reviewer: ```suggestion
  Its features and attribute table depend on the selected output matrix type..
```

.. _qgisdistancetonearesthublinetohub:

Distance to nearest hub (line to hub)
-------------------------------------

> Creates a line that joins each feature of an input vector to the nearest feature

Reviewer: ```suggestion
Creates lines that join each feature of an input vector to the nearest feature
```

in a destination layer. Distances are calculated based on the :ref:`center
<qgispointonsurface>` of each feature.

.. figure:: img/distance_hub.png
  :align: center

  Display the nearest hub for the red input features

Parameters
..........

``Source points layer`` [vector: any]
  Vector layer for which the nearest feature is searched.

``Destination hubs layer`` [vector: any]

>   Vector layer containing the points to search the nearest within.

Reviewer: ```suggestion
  Vector layer containing the features to search for.
```

Outputs
.......
``Hub distance`` [vector: line]
 Line vector layer with the attributes of the input features, the identifier
 of their closest feature and the calculated distance.

.. _qgisdistancetonearesthubpoints:

Distance to nearest hub (points)
--------------------------------
Creates a point layer representing the :ref:`center <qgispointonsurface>` of the
input features with the addition of two fields containing the identifier of the
nearest feature (based on its center point) and the distance between the points.

Parameters
..........

``Source points layer`` [vector: any]
  Vector layer for which the nearest feature is searched.

``Destination hubs layer`` [vector: any]

>   Vector layer containing the points to search the nearest within.

Reviewer: ```suggestion
  Vector layer containing the features to search for.
```
